### PR TITLE
OriginalMap: add map preview

### PR DESCRIPTION
**Pull request by tomsoftware**
- read and return map preview from original map file
- disable checksum validation cause of speed problems:
=> Add "disable-original-maps-checksum = true" to options.prp
- bugfixes

Example of original-maps displaying within new GUI from andreasb242/codingberlin :
ori_map_preview

PS: your new GUI looks quite good :-)

### CODE REVIEW

File: jsettlers.logic/src/jsettlers/logic/map/original/OriginalMapFileContentReader.java [hunk 1]
```diff
@@ -322,6 +330,57 @@ public String readMapQuestText() {
...
+		// - height and width are the same
+		int wh = readBEWordFrom(pos);
+		pos+=2;
+		int unknown = readBEWordFrom(pos);
+		pos+=2;
+		
+		float scale_width = wh / width;
+		float scale_height = wh / height;
+		
+		int out_index = 0;
```
Comment (codingberlin) [inline jsettlers.logic/src/jsettlers/logic/map/original/OriginalMapFileContentReader.java]: Please write all your variable names in camel case e.g. ```out_index```-> ```outIndex```

File: jsettlers.logic/src/jsettlers/logic/map/original/OriginalMapFileContentReader.java [hunk 2]
```diff
@@ -322,6 +330,57 @@ public String readMapQuestText() {
 		return mapQuestText;
 	}
 
+	public short[] getPreviewImage(int width, int height)
+	{
```
Comment (codingberlin) [inline jsettlers.logic/src/jsettlers/logic/map/original/OriginalMapFileContentReader.java]: Please move all the open curly brackets to the end of the method name line.

**Comment by codingberlin** [comment]
This is awesome! Quite funny that this are exactly the two things I was looking for in the last days while implementing the new join game screen:
* to not wait so long when I'm rerunning the GUI to see my recent code changes
* nice preview map images

(I add just two minor code style comments)

Thx a lot!

**Comment by tomsoftware** [comment]
+Add image cache
 + codingberlin: sorry for my bad style ;-)

=> I'm not sure if it is good to remove the checksum validation. Without the only validation if this is the right map-format is the file-extention and https://github.com/tomsoftware/settlers-remake/blob/master/jsettlers.logic/src/jsettlers/logic/map/original/OriginalMapFileContentReader.java#L205
(FYI: Settler 4 uses another number - so this should be fine) but this is not a very strong validation!

One last Problem:
If the OriginalMapLoader() catches an Error cause the map-file is in the wrong format, how should the constructor behave to prevent the caller from showing/using this map-file?
?  throws new IOException (); but this is the wrong Error-MSG and it needs to be caught by the caller...
https://github.com/tomsoftware/settlers-remake/blob/master/jsettlers.logic/src/jsettlers/logic/map/original/OriginalMapLoader.java#L56
and
https://github.com/tomsoftware/settlers-remake/blob/master/jsettlers.logic/src/jsettlers/logic/map/original/OriginalMapLoader.java#L64

**Comment by andreasb242** [comment]
>PS: your new GUI looks quite good :-)

Thank you ;-)

Did you see my pull requests  which fixes the enconding of the description?
https://github.com/jsettlers/settlers-remake/pull/273

For maps with errors are two possibilities:
1. Simple ignore them, log the error when we have a logging system
2. Display an error to the user: I added an error dialog, which can be called. You can find it on my branch or on codingberlin's branch.

https://github.com/andreasb242/settlers-remake/tree/Swing-Look-and-Feel/jsettlers.main.swing/src/jsettlers/exceptionhandler

But my preferred way would be to ignore the map.
Here a screenshot of the Editor: The user should create a bug report and copy the stacktrace.

So if you think you need the information what's wrong with the map: Put all information you need to *one* Exception and call the Exceptionhandler with this single exception...

bildschirmfoto vom 2016-01-16 12 07 50

**Comment by tomsoftware** [comment]
@andreasb242: yes thanks, I forgot about  the different codepages... I've used your pull #273 but did some code cleanup and rewrite this part again...

Now I set the behavior on map loading errors -> ignore map:
If a map can't by loaded due to a wrong Checksum or wrong File-Format (e.g. remake-map with ".map" extension) the OriginalMapLoader.getFileHeader() returns NULL, no exeption is called and MapList.foundMap() hides the map from the list. (only a msg on System.err.() is printed)

**Comment by andreasb242** [comment]
@tomsoftware So I should close my pull request now, right?

**Comment by tomsoftware** [comment]
@andreasb242 : yes...

**Comment by andreasb242** [comment]
Ok, Closed

**Comment by codingberlin** [comment]
I reviewed this pull request. Please merge the latest changes from master into this branch then I will accept it.

**Comment by tomsoftware** [comment]
@codingberlin : done!

it was me a pesurse to help you/this project :-) - Thanks!

**Comment by andreas-eberle** [comment]
@tomsoftware: Thanks to you for contributing!

**Pull request merged**